From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Haoran Wang <elven.wang@nxp.com>
Date: Thu, 03 Apr 2026 00:00:00 +0800
Subject: [PATCH] Fix GCC 15 warnings in MATTER_MEDIA code

Fix compilation errors when building with GCC 15 and -Werror:

1. Fix format specifier: strlen() returns size_t, use %zu instead of %d
2. Fix unused return value of write() with warn_unused_result attribute  
3. Fix unused return value of read() with warn_unused_result attribute
4. Fix unused variable stdin_poll when MATTER_MEDIA is defined

These issues were introduced by the MATTER-1573 patch and only manifest
with GCC 15's stricter warning checks.

Signed-off-by: Haoran Wang <elven.wang@nxp.com>
Upstream-Status: Inappropriate [GCC 15 compatibility fix for platform patch]
---
 tools/gplay2/gplay2.c | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/tools/gplay2/gplay2.c b/tools/gplay2/gplay2.c
index 72a373a..b1c2d3e 100755
--- a/tools/gplay2/gplay2.c
+++ b/tools/gplay2/gplay2.c
@@ -120,8 +120,11 @@ char * media_fifo_ack = "/tmp/fifo_nxp_media_ack";
 void
 ack_matter(const gchar * ack)
 {
+  ssize_t ret;
   int fd = open(media_fifo_ack, O_WRONLY);
-  write(fd, ack, strlen(ack)+1);
+  ret = write(fd, ack, strlen(ack)+1);
+  if (ret < 0)
+    g_warning("ack_matter: write failed");
   sleep(1);
   close(fd);
 }
@@ -1043,7 +1046,9 @@ input_thread_fun (gpointer data)
   GstPlay *play = sPlay->play;
   GstPlayVideoRenderer *VideoRender = sPlay->VideoRender;
   gplay_pconfigions *options = sPlay->options;
+#ifndef MATTER_MEDIA
   struct pollfd stdin_poll = { STDIN_FILENO, POLLIN|POLLPRI };
+#endif
 
   while (gexit_input_thread == FALSE) {
     sCommand[0] = ' ';
@@ -1052,10 +1057,13 @@ input_thread_fun (gpointer data)
 #ifdef MATTER_MEDIA
     fd = open(media_fifo, O_RDONLY);
-    read(fd, sCommand, sizeof(sCommand));
+    {
+      ssize_t bytes_read = read(fd, sCommand, sizeof(sCommand));
+      (void)bytes_read;  /* suppress warning - we check strlen instead */
+    }
     close(fd);
     ack_matter("ack");
     if (strlen(sCommand) != 0) {
-        g_print("get Matter command(%d): %s\n", strlen(sCommand),sCommand);
+        g_print("get Matter command(%zu): %s\n", strlen(sCommand),sCommand);
     }
 #else
     if (poll (&stdin_poll, 1, 500)) {
-- 
2.34.1
